analysis/token/synonym/synonym.go

```diff
@@ -0,0 +1,332 @@
+		if err != nil {
+			return nil, err
+		}
+		for _, i := range edges {
+			if i != SeparatingCharacter {
+				newCur, newOut := fst.AcceptWithVal(curr, i)
+				newCurWord := curWord + string(i)
+				fuzzyQueue, err = fuzzySearchFST(output+newOut, newCur, depth-1,
+					fuzzyQueue, fst, newCurWord, matchTry, fuzziness)
+				if err != nil {
+					return nil, err
+				}
+			}
+		}
+	}
+	return fuzzyQueue, nil
+}
+
+func checkForMatch(tokenPos int, input analysis.TokenStream, keepOrig *bool,
+	fst *vellum.FST, vellumStructMap map[uint64][]uint64, synTokenEnd *int,
+	fuzziness int, prefix int) (int, []uint64, analysis.TokenStream, error) {
+
+	maxMatchedTokenPos := tokenPos + 1
+	fuzzyQueueLen := 1
+	var matchedSynonyms, seenSynonyms []uint64
+	var consumedTokens, seenTokens analysis.TokenStream
+	var fuzzyQueue, queueWithoutDeadEnds []fuzzyStruct
+	var currOutput, tmp uint64
+	var tokenLen, matchLen, queuePos, curr, newCurr int
+	var matched, isMatch bool
+	var err error
+	fuzzyQueue = append(fuzzyQueue, fuzzyStruct{
+		state:  fst.Start(),
+		output: 0,
+	})
+	for tokenPos != len(input) {
+		queuePos = 0
+		for queuePos < fuzzyQueueLen {
+			tokenLen = len(input[tokenPos].Term)
+			matchLen = 0
+			var curWord []byte
+			curr = fuzzyQueue[0].state
+			currOutput = fuzzyQueue[0].output
+			fuzzyQueue = fuzzyQueue[1:]
+			for _, character := range input[tokenPos].Term {
+				newCurr, tmp = fst.AcceptWithVal(curr, character)
+				if newCurr == 1 {
+					if fuzziness == 0 {
+						return maxMatchedTokenPos, matchedSynonyms,
+							consumedTokens, nil
+					} else if matchLen < prefix {
+						break
+					} else {
+						fuzzyQueue, err = fuzzySearchFST(currOutput, curr,
+							tokenLen-matchLen+fuzziness, fuzzyQueue, fst,
+							string(curWord), string(input[tokenPos].Term),
+							fuzziness)
+						if err != nil {
+							return -1, nil, nil, err
+						}
+						break
+					}
+				} else {
+					matchLen++
+					curr = newCurr
+					currOutput += tmp
+					curWord = append(curWord, character)
+				}
+			}
+			if matchLen == tokenLen {
+				wholeWord := fst.Accept(curr, SeparatingCharacter)
+				numEdges, err := fst.GetNumTransitionsForState(curr)
+				if err != nil {
+					return -1, nil, nil, err
+				}
+				if numEdges == 0 || wholeWord != 1 {
+					fuzzyQueue = append(fuzzyQueue, fuzzyStruct{
+						output: currOutput,
+						state:  curr,
+					})
+				} else {
+					if fuzziness == 0 {
+						return maxMatchedTokenPos, matchedSynonyms,
+							consumedTokens, nil
+					} else if matchLen < prefix {
+						break
+					} else {
+						fuzzyQueue, err = fuzzySearchFST(currOutput, curr,
+							fuzziness, fuzzyQueue, fst, string(curWord),
+							string(input[tokenPos].Term), fuzziness)
+						if err != nil {
+							return -1, nil, nil, err
+						}
+					}
+				}
+			}
+			queuePos += 1
+		}
+		seenTokens = append(seenTokens, input[tokenPos])
+		tokenPos++
+		seenSynonyms = nil
+		queueWithoutDeadEnds = nil
+		matched = false
```

Comment:
I don't see any modifications to these variable above, so it would be false/nil by default because of the initialization on lines 73 -79, please clean this up. I see something